server/reports/utils.js

```diff
@@ -0,0 +1,30 @@
+const cleanUpNodes = (nodes) => {
+  let cleanNodes = [];
+  nodes.forEach((node) => {
+    if (Object.keys(node).length != 0) {
+      cleanNodes.push(cleanUpObject(node));
+    }
+  });
+  return cleanNodes;
+};
+
+const cleanUpObject = (node) => {
+  let newNode = {};
+  // remove empty keys which will fail to parse
+  Object.keys(node).forEach(function (key) {
+    if (node[key] !== "" && node[key] !== undefined && node[key] !== null) {
+      if (typeof node[key] === "object") {
+        // recursively remove empty strings or undefined from graphql query
+        newNode[key] = cleanUpObject(node[key]);
+      } else {
+        newNode[key] = node[key];
+      }
+    }
+  });
+  return newNode;
+};
+
+module.exports = {
+  cleanUpObject,
+  cleanUpNodes,
+};
```

Comment:
Have added this to remove empty strings from json objects which will fail to parse. However, tera will throw errors if it tries to render non existent values.

I'm not sure if it is likely that we will return non existent values from queries, but might be safe to use 'ignore missing' in tera to handle this